pkg/apis/config/v1alpha2/meshrootcertificate.go

```diff
@@ -112,15 +112,10 @@ type TresorCASpec struct {
 }
 
 // MeshRootCertificateStatus defines the status of the MeshRootCertificate resource
-type MeshRootCertificateStatus struct {
+type ProviderState struct {
 	// State specifies the state of the root certificate rotation
+	// All states are specified in constants.go
 	State string `json:"state"`
```

Comment:
I agree with Seans comment. If there is a name change I think `state` could be updated to `providerState`. Since the MRC will contain more information than just the provider I think it makes sense to be specific about what component the state is referencing.